### DISCUSSION TITLE
New layer for activities and filters for type of activity, difficulty..etc to display

### DISCUSSION THREAD

**Discussion by DeathfireD** [organicmaps/organicmaps#3264] [Ideas]
I have a few ideas. Some that may have already been suggested. I noticed OSM allows you to add/modify trails and label them for use as hiking, biking, horse...etc. They also allow you to attach a wide variety of other useful info to the trail. Like the trail's difficulty and condition.  

It would be nice to have a new layer called "Activities". When this layer is turned on it would highlight every trail and make them clickable. When clicking on them Organic Maps would display the OSM raw label information in a way that's similar to the wiki popup. Each OSM label would be clickable so when clicked, it would go to that wiki page on OSM to explain what it's values mean.

Adding a new filter system to just this "Activities" layer would also be very helpful too. A person looking to Mountain Bike could select the activity "Mountain Biking" and Organic Maps would only highlight trails that support that activity. Like above, they would be able to click on just those trails to view the raw OSM data and even "favorite" the trail for future riding.

I think adding a new layer for ALL activities might slow down Organic Maps so it might be a good idea to limit it to just display 1 or 2 popular activities by default. Like Hiking. Then the user can go into filters to adjust what should display.

Here's a few things I found interesting on OSM.
https://wiki.openstreetmap.org/wiki/Hiking#Offline-Maps
https://wiki.openstreetmap.org/wiki/Hiking_maps
https://wiki.openstreetmap.org/wiki/Key:sac_scale
https://wiki.openstreetmap.org/wiki/Key:mtb:scale?uselang=en

**Comment by biodranik** [comment]
Thanks, the idea is good, I think there was already a similar issue on Github. It would be great to collect and prepare the information for the implementation:
1. The list of the most important tags and their combinations to support.
2. Links to real examples in the OSM
3. Ideas about highlighting, rendering, using, searching, and following these trails. Maybe screenshots from other apps.

**Comment by DeathfireD** [comment]
1. I'd assume mtb:scale:imba and mbv:scale would be the most important for bikers and sac_scale and trail_visibility would be the most important for hikers. As a hiker I really just want to know what the difficulty/classification of a trail is. Unfortunately since OSM user submitted date is inconsistent, most trails don't have ratings or people put them in the name of the trail (ex. The Loop Trail **(Blue)** ). In any case, if the label/tags are there, It would be nice to see them and also be able to filter based on Hiking, Biking, Horse. That way it'll encourage others to go add the information to the trails.
https://wiki.openstreetmap.org/wiki/Key:mtb:scale:imba?uselang=en
https://wiki.openstreetmap.org/wiki/Key:mtb:scale?uselang=en
https://wiki.openstreetmap.org/wiki/Key:sac_scale
https://wiki.openstreetmap.org/wiki/Key:trail_visibility

2. Here's an example of how OSM displays the label/tag information for a trail off to the left on PC. If you go to layers and turn on Map Data it allows you to click on trails and objects to view the label/tag information for that trail. This is similar to what I'm suggesting. The only difference is Organic Maps should limit it to just trails when the proposed "Activities" layer is turned on. This way the map isnt cluttered with information about EVERY object like rocks and ponds..etc. https://www.openstreetmap.org/way/44947391#map=17/44.25540/-71.32280&layers=D

3. Here is another link to a website that uses OSM info and allows people to click on hiking trails https://hiking.waymarkedtrails.org/#route?id=13125079. When clicking on them it highlights them in yellow and displays all the trail tag/label info off to the right. This would be how i'd suggest displaying the raw info at the bottom of the info popup. Just a simple table with all tags/labels and their values. Make them clickable so a person can click them and go to the OSM wiki to learn more about that tag.

Here are some screenshots on how the Mountain Project App handles trails. When zoomed out it will display a circle with the number of trails in that area. When clicking on that number it zooms in. When zoomed in they have all of the trails distinguishable via the color of difficulty that they've been designated and when you click on one, it adds a glow effect around the trail with a start and end pin to indicate that you've clicked on it. It also displays info on the bottom about the trail. Obviously this info is not from OSM though so you'd probably just display the info about the elevation, length, est time to hike, and label/tags under it. This however is a perfect example of what I'm suggesting.
 
![TrailsWithInfo2](https://user-images.githubusercontent.com/1296196/187053500-214840d4-fcdb-4fb9-8100-36b5fd1a37a5.jpg)
![CombinedTrails](https://user-images.githubusercontent.com/1296196/187053509-d3a68be2-f7dd-41cd-8590-282ce724c3fb.jpg)
![HighlightedTrailsWithColorDifficulty](https://user-images.githubusercontent.com/1296196/187053514-0d72d40a-fb4b-4c77-a3eb-37deb129680e.jpg)


Navigating is already pretty solid. As a hiker/mountain biker I really just want an easier way to distinguish and get info about the trails. The only suggestion I'd give about the navigation system is to allow people the ability add in way points so they can plot out a custom path that spans over multiple trails instead of having to import a file. Right now Organic maps allows you to use "stops" however sometimes it gets confused and doesn't quite work the way you'd expect it to causing you to loop back in some parts. Not a huge deal since Organic Maps is all offline and we can just adjust our path accordingly when we reach the end of a trail though.
